spec/models/account_migration_spec.rb

```diff
@@ -0,0 +1,158 @@
+      expect(account_migration).to receive(:perform!)
+      account_migration.receive
+    end
+  end
+
+  describe "sender" do
+    context "with remote old user" do
+      include_context "with remote old user"
+
+      it "creates ephemeral user when private key is provided" do
+        account_migration.old_private_key = old_user.serialized_private_key
+        sender = account_migration.sender
+        expect(sender.id).to eq(nil)
+        expect(sender.diaspora_handle).to eq(old_user.diaspora_handle)
+        expect(sender.encryption_key.to_s).to eq(old_user.encryption_key.to_s)
+      end
+
+      it "raises when no private key is provided" do
+        expect {
+          account_migration.sender
+        }.to raise_error("can't build sender without old private key defined")
+      end
+    end
+
+    context "with local old user" do
+      include_context "with local old user"
+
+      it "matches the old user" do
+        expect(account_migration.sender).to eq(old_user)
+      end
+    end
+  end
+
+  describe "performed?" do
+    it "is changed after perform!" do
+      expect {
+        account_migration.perform!
+      }.to change(account_migration, :performed?).to be_truthy
+    end
+
+    it "calls old_person.closed_account?" do
+      expect(account_migration.old_person).to receive(:closed_account?)
+      account_migration.performed?
+    end
+  end
+
+  context "with local new user" do
+    include_context "with local new user"
+
+    describe "subscribers" do
+      it "picks remote subscribers of new user profile and old person" do
+        _local_friend, remote_contact = DataGenerator.create(new_user, %i[mutual_friend remote_mutual_friend])
+        expect(account_migration.new_person.owner.profile).to receive(:subscribers).and_call_original
+        expect(account_migration.subscribers).to match_array([remote_contact.person, old_person])
+      end
+
+      context "with local old user" do
+        include_context "with local old user"
+
+        it "doesn't include old person" do
+          expect(account_migration.subscribers).to be_empty
+        end
+      end
+    end
+  end
+
+  describe "perform!" do
+    let(:embedded_account_deleter) { account_migration.send(:account_deleter) }
+
+    it "raises if already performed" do
+      expect(account_migration).to receive(:performed?).and_return(true)
+      expect {
+        account_migration.perform!
+      }.to raise_error("already performed")
+    end
+
+    it "calls AccountDeleter#tombstone_person_and_profile" do
+      expect(embedded_account_deleter).to receive(:tombstone_person_and_profile)
+      account_migration.perform!
+    end
+
+    # TODO: This test is very slow. Also it is duplicated in the integration test.
+    # TODO: Should we remove it or leave? It's possible to optimize the test in a manner
+    # TODO: like account deletion integration test work. It'll make it a slightly faster.
+    # it_behaves_like "it updates person references" do
```

Comment:
Well, I think it is a better approach anyway, than mocking a private method, since tests shouldn't rely on private methods. Private methods are supposed to be allowed to change freely the way they work and named. While there is nothing wrong in having a very special public method (we have many very special scopes which are used only once). And this will allow us to write a proper test without relying on private methods and checking attributes directly as in the integration test.

I think that we could make a helper module encapsulating this `model.find(old_id).send(association).update_all(key_id => new_id)` behavior, and include it to our `AccountMigration` model.  We can make this module publiclty accessible via a scope inside `AccountMigration` model to give an idea that this behavior is account migration specific and isn't expected to be used outside. Something like `AccountMigration::ReferenceUpdater`. What do you think?